**Loading the Dataset.**

In [1]:
#Loading the rsics_dataset and converting it from tar.gz to csv file format.
import tarfile
import csv

#open the rsics_dataset.tar.gz output_file
with tarfile.open('rsics_dataset.tar.gz', 'r:gz') as tar:
  #extract and process each file
  for member in tar.getmembers():
    #checking for any csv files in the dataset and processing accordingly if not
    if member.name.endswith('.csv'):
      #Extract the file into a csv format
      file = tar.extractfile(member)
      #Read the data in the csv file
      csv_data = file.read().decode('utf-8') #Assuimg that the possible text in the data is encoded in utf-08
      #Extract the csv data to a new csv file
      with open ('rsics.csv', 'a') as rsics_file:
        rsics_file.write(csv_data)

print('converted rsics file')




converted rsics file


In [5]:
import pandas as pd
#file path to the csv file
csv_file_path = 'rsics.csv'

#Load the dataset
rsics_data= pd.read_csv(csv_file_path)

#Display info of the dataset
print('Rsics_Dataset_info:')
print(rsics_data.info())

#Display the first few rows of the dataset.
print('\nFirst few rows of the dataset:')
print(rsics_data.head())



<ipython-input-5-7315aa498396>:6: DtypeWarning: Columns (12,13) have mixed types. Specify dtype option on import or set low_memory=False.
  rsics_data= pd.read_csv(csv_file_path)


Rsics_Dataset_info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 242605 entries, 0 to 242604
Data columns (total 14 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Dataset           242541 non-null  object
 1   Partition         242541 non-null  object
 2   SentenceID        242285 non-null  object
 3   Threshold         242285 non-null  object
 4   MergedSelections  242285 non-null  object
 5   Unselected        242217 non-null  object
 6   Selected          234751 non-null  object
 7   Greeting          208111 non-null  object
 8   Backstory         242285 non-null  object
 9   Justification     178283 non-null  object
 10  Rant              178283 non-null  object
 11  Gratitude         178283 non-null  object
 12  Other             13519 non-null   object
 13  Express Emotion   13519 non-null   object
dtypes: object(14)
memory usage: 25.9+ MB
None

First few rows of the dataset:
  Dataset Partition SentenceID Thresho

Number of Columns- 14

Number of Rows- 242605

Data Type is object meaning it consists of mixed datatypes.

**DATA PREPROCESSING**

Checking the count of missing values, outliers and duplicates.

In [7]:
#Checking the count of missing values, outliers and duplicates
#load rsics.csv file into df
df= pd.read_csv('rsics.csv')

#check for missing values
missing_values = df.isnull().sum()
print('Missing Values:\n', missing_values)

#check for duplicates
duplicate_rows = df[df.duplicated()]
print('Duplicate Rows:\n', duplicate_rows)

#check for outliers

# Check for outliers
# You can use statistical methods or visualization techniques here

# Check for duplicates
duplicate_rows = df[df.duplicated()]
print("Duplicate Rows:\n", duplicate_rows)

#check the data distribution
df.describe()


<ipython-input-7-4a7f6c19cad8>:3: DtypeWarning: Columns (12,13) have mixed types. Specify dtype option on import or set low_memory=False.
  df= pd.read_csv('rsics.csv')


Missing Values:
 Dataset                 64
Partition               64
SentenceID             320
Threshold              320
MergedSelections       320
Unselected             388
Selected              7854
Greeting             34494
Backstory              320
Justification        64322
Rant                 64322
Gratitude            64322
Other               229086
Express Emotion     229086
dtype: int64
Duplicate Rows:
                   Dataset       Partition   SentenceID   Threshold  \
8712       Annotator A ID  Annotator B ID  Parition ID  Corpora ID   
11506                 NaN             NaN          NaN         NaN   
11511      Annotator A ID  Annotator B ID  Parition ID  Corpora ID   
14248                 NaN             NaN          NaN         NaN   
14253      Annotator A ID  Annotator B ID  Parition ID  Corpora ID   
...                   ...             ...          ...         ...   
242600                NaN             NaN          NaN         NaN   
242601     Aver

Dataset Partition SentenceID Threshold MergedSelections  \
count   242541    242541     242285    242285           242285   
unique      15       115       8005      1235            13410   
top          1         1          2         2                1   
freq     40352     60352      81408     56694            16596   

                                               Unselected Selected Greeting  \
count                                              242217   234751   208111   
unique                                              23981    32665    27276   
top     How many award miles do I need for a Mileagepl...        0        0   
freq                                                 2140     2370    13588   

       Backstory Justification    Rant Gratitude    Other  Express Emotion  
count     242285        178283  178283    178283  13519.0          13519.0  
unique      8302           847    7735         5      5.0              5.0  
top            0             0       1         1      0.0              0.0  
freq       45164         95966   81928    118060   5315.0           6639.0